agreement/persistence.go

```diff
@@ -33,10 +33,14 @@ import (
 
 // diskState represents the state required by the agreement protocol to be persistent.
 type diskState struct {
-	Router, Player, Clock []byte
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
```

Comment:
omitempty doesn't change the ability to deserialize fields as long as the fields could ever be non empty. It just changes serialization. 

The only problem that I saw was with rootRouter which has proposalManager and voteAggregator fields which are provably empty and therefore don't get methods generalized for them by the msgp generator.

Reply:
> I think we could probably leave off decorating all the structs with `_struct` fields and things would be fine/better? This is assuming protocol.EncodeReflect does not implement the omitempty behavior

We do need to include the `_struct` but it can just be `_struct struct `codec:","` if we don't want omitempty but they don't seem to be harmful for things other than re-encoding to the exact same value. Decoding is not affected

Reply:
I've removed the omitempty's and performance hasn't changed. Marking as resolved